torch/jit/__init__.py

```diff
@@ -581,9 +582,16 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
-def script(fn, optimize=True, _frames_up=0):
+def weak_script(fn):
+    fn._jit_is_weak_script = True
+    return fn
+
+
+def script(fn, optimize=True, _frames_up=0, _is_weak=False):
```

Comment:
It is more orthogonal to keep the implementation of `script` as-is and make the pathway to weak script separate. The general "code smell" issue that shows up here is that there is a boolean flag that is being checked multiple times inside this function. In these cases, it is often better to write a separate function.